setup/deploy_nfd_test.go

```diff
@@ -98,28 +103,40 @@ var _ = Describe("deploy_nfd_operator :", Ordered, func() {
 			},
 			ObjectMeta: metav1.ObjectMeta{
 				Namespace: internal.Config.NameSpace,
-				Name:      "nfd-cr-testing",
+				Name:      nfdCrName,
 			},
 		}
 
-		resp, err := ocputils.CreateDynamicResource(config, nfdv1.GroupVersion.WithResource("nodefeaturediscoveries"), nfd, nfd.Namespace)
+		resp, err := ocputils.CreateDynamicResource(config, nfdv1.GroupVersion.WithResource(nfdResource), nfd, nfd.Namespace)
 		Expect(err).ToNot(HaveOccurred())
 		var respNfd nfdv1.NodeFeatureDiscovery = nfdv1.NodeFeatureDiscovery{}
 		err = runtime.DefaultUnstructuredConverter.FromUnstructured(resp.UnstructuredContent(), &respNfd)
 		Expect(err).ToNot(HaveOccurred())
-		err = testutils.SaveAsJsonToArtifactsDir(respNfd, "nfd_cr.json")
+		err = testutils.SaveAsJsonToArtifactsDir(respNfd, "nfd_cr_create_response.json")
 		Expect(err).ToNot(HaveOccurred())
 
 	})
 
-	It("wait for NFD labels", func() {
+	It("wait for NFD labels and capture nfd cr state", func() {
 		err := testutils.ExecWithRetryBackoff("wait for NFD labels", func() bool {
 			nodes, err := ocputils.GetNodesByLabel(config, "feature.node.kubernetes.io/system-os_release.ID=rhcos")
 			if err != nil {
 				return false
 			}
 			return len(nodes.Items) > 0
 		}, 20, 30*time.Second)
+
+		if err != nil {
+			testutils.Printf("error", "Failed to find labels on nodes. %v", err)
+		}
+
+		// Regarless if successful, we want to have the CR in artifacts
+		nfdCr := &nfdv1.NodeFeatureDiscovery{}
+		e := ocputils.GetDynamicRecource(config, nfdv1.GroupVersion.WithResource(nfdResource), internal.Config.NameSpace, nfdCrName, nfdCr)
```

Comment:
Looks like a type in the ocputils package

Reply:
👍🏽  Updated